examples/04-Retrieval-Model.ipynb

```diff
@@ -0,0 +1,1866 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 9,
+   "id": "ab665acc",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "model = mm.TwoTowerModel(\n",
+    "    schema,\n",
+    "    query_tower=mm.MLPBlock([128, 64], no_activation_last_layer=True),        \n",
+    "    loss=\"categorical_crossentropy\",  \n",
+    "    samplers=[mm.InBatchSampler()],\n",
+    "    embedding_options = mm.EmbeddingOptions(infer_embedding_sizes=True),\n",
+    "    metrics=[mm.RecallAt(10), mm.NDCGAt(10)]\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "2480a7fa",
+   "metadata": {},
+   "source": [
+    "Let's explain the parameters in the TwoTowerModel():\n",
+    "- no_activation_last_layer: when set True, no activation is used for top hidden layer. Learn more [here](https://storage.googleapis.com/pub-tools-public-publication-data/pdf/b9f4e78a8830fe5afcf2f0452862fb3c0d6584ea.pdf).\n",
+    "- infer_embedding_sizes: when set True, automatically defines the embedding dimension from the feature cardinality in the schema\n",
+    "\n",
+    "**Metrics:**\n",
+    "\n",
+    "The following information retrieval metrics are used to compute the Top-10 accuracy of recommendation lists containing all items:\n",
+    "\n",
+    "- **Normalized Discounted Cumulative Gain (NDCG@10)**: NDCG accounts for rank of the relevant item in the recommendation list and is a more fine-grained metric than HR, which only verifies whether the relevant item is among the top-k items.\n",
+    "\n",
+    "- **Recall@10**: Also known as HitRate@n when there is only one relevant item in the recommendation list. Recall just verifies whether the relevant item is among the top-n items."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "a65948b0",
+   "metadata": {},
+   "source": [
+    "We need to initialize the dataloaders."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 10,
+   "id": "a7525cea",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "/usr/local/lib/python3.8/dist-packages/cudf/core/dataframe.py:1253: UserWarning: The deep parameter is ignored and is only included for pandas compatibility.\n",
+      "  warnings.warn(\n"
+     ]
+    }
+   ],
+   "source": [
+    "batch_size = 4096\n",
+    "\n",
+    "train = Dataset(os.path.join(output_path + '/train/*.parquet'), part_size=\"500MB\")\n",
+    "valid = Dataset(os.path.join(output_path + '/test/*.parquet'), part_size=\"500MB\")"
```

Comment:
The os.path.join call here isn't doing anything, since you're already joining with string concat with the `+`. How about 
```suggestion
    "train = Dataset(os.path.join(output_path, 'train', '*.parquet'), part_size=\"500MB\")\n",
    "valid = Dataset(os.path.join(output_path, 'test', '*.parquet'), part_size=\"500MB\")"
```